here-naksha-lib-core/src/test/java/com/here/naksha/lib/core/util/diff/PatcherTest.java

```diff
@@ -57,6 +59,88 @@ void basic() throws IOException {
     assertNull(newDiff);
   }
 
+  @Test
+  void testCompareBasicNestedJson() throws JSONException {
+    final JsonObject f3 =
+            JsonSerializable.deserialize(IoHelp.readResource("patcher/feature_3.json"), JsonObject.class);
+    assertNotNull(f3);
+    final JsonObject f4 =
+            JsonSerializable.deserialize(IoHelp.readResource("patcher/feature_4.json"), JsonObject.class);
+    assertNotNull(f4);
+
+    final Difference diff34 = Patcher.getDifference(f3, f4);
+    assertNotNull(diff34);
+    assert (diff34 instanceof MapDiff);
+
+    // Assert outermost layer
+    final MapDiff mapDiff34 = (MapDiff) diff34;
+    // TODO if possible to serialize Difference, simply compare the serialized Difference object with test file content
+    assertTrue(mapDiff34.get("isAdded") instanceof InsertOp);
+    assertTrue(mapDiff34.get("willBeUpdated") instanceof UpdateOp);
+    assertTrue(mapDiff34.get("firstToBeDeleted") instanceof RemoveOp);
+    assertTrue(mapDiff34.get("map") instanceof MapDiff);
+    assertTrue(mapDiff34.get("array") instanceof ListDiff);
+
+    // Assert nested layer
+    final MapDiff nestedMapDiff34 = (MapDiff) mapDiff34.get("map");
+    // "mapID" is retained, does not appear in nestedMapDiff34
+    assertTrue(nestedMapDiff34.get("isAdded") instanceof InsertOp);
+    assertTrue(nestedMapDiff34.get("willBeUpdated") instanceof UpdateOp);
+    assertTrue(nestedMapDiff34.get("willBeDeleted") instanceof RemoveOp);
+
+    // Assert nested array
+    final ListDiff nestedArrayDiff34 = (ListDiff) mapDiff34.get("array");
+    assertTrue(nestedArrayDiff34.get(1) instanceof UpdateOp);
+    assertTrue(nestedArrayDiff34.get(2) instanceof MapDiff);
+    // "retainedElement" is retained, does not appear in nestedMapDiff34
+    // InsertOp case for array (ListDiff) is addressed in the test testCompareSameArrayDifferentOrder()
+    assertTrue(nestedArrayDiff34.get(3) instanceof RemoveOp);
+
+    // Some extra nested JSON object in array assertions
+    assertTrue(((MapDiff) nestedArrayDiff34.get(2)).get("isAddedProperty") instanceof InsertOp);
+    assertTrue(((MapDiff) nestedArrayDiff34.get(2)).get("willBeDeletedProperty") instanceof RemoveOp);
+
+    // Modify the whole difference to get rid of all RemoveOp
```

Comment:
Trying to understand, why do we remove the diff before applying patch?

Reply:
No hard target actually, the normal patching from one feature to the other has already been covered in the basic() test, so this is just to simulate the usage of the patcher in the PATCH REST API.

Reply:
Ok